src/pkg/artifact/controller.go

```diff
@@ -13,3 +13,103 @@
 // limitations under the License.
 
 package artifact
```

Comment:
As we discussed, the controller will be in the `api/artifact` folder.

I'll complete the programming model tomorrow.